### PR TITLE
Nicotine is now addictive again, adds nicotine withdrawal effects.

### LINKED ISSUE #57631: Nicotine is completely nonaddictive
## Round ID:

Any

## Testmerges:

Doesn't matter

## Reproduction:

Take smoker quirk. Suffer absolutely no withdrawal symptoms. Enjoy your four free quirk points.

### PR DESCRIPTION
**Pull request by prodirus**
## About The Pull Request

Fixes #57631 

Nicotine didn't have an `addiction_types` value added to it after the addiction rework here: #56923

About 9u of pure liquid Nicotine or 2 Robust cigarettes back to back should cause addiction.

Added new nicotine withdrawal effects as the existing withdrawal effects did not seem like they fit well enough. What's there is still pretty boring though, so I'm open for suggestions!

## Why It's Good For The Game

Smokers can properly get their fix now!

## Changelog
:cl:
fix: Nicotine is addictive again.
add: New nicotine withdrawal effects. Time to feel miserable and jittery if you don't get your fix!
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 1]
```diff
@@ -45,6 +45,7 @@
 	metabolization_rate = 0.125 * REAGENTS_METABOLISM
 	ph = 8
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	addiction_types = list(/datum/addiction/nicotine = 50) // pretty dang addictive
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm:48]: 👀 

time to use incredibly tiny microdoses of nicotine to get people addicted to cigarettes

Comment (ATH1909) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:48]: like, for reference here, addiction_gain_threshold is 600, and nicotine has an incredibly low metabolization rate. correct me if I'm wrong here, but I think that even 0.5u of nicotine should be well more than enough to get you addicted to nicotine.

Comment (prodirus) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:48]: Yeah, I've been playing around with the numbers, but I also want people to be able to get addicted from normal smoking in a reasonable amount of time. With the numbers now you need to smoke basically an entire cigarette non-stop to get addicted.

Comment (prodirus) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:48]: Yeah, 50 is way too high, going to tone it down.

Comment (prodirus) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Right, should be a bit more sensible now. 9u of nicotine or 2 Robust cigarettes back to back will cause addiction at neutral mood.

File: code/modules/reagents/withdrawal/generic_addictions.dm [hunk 2]
```diff
@@ -228,3 +228,26 @@
...
+	withdrawal_stage_messages = list("Feel like having a smoke...", "Can't concentrate. Really need a smoke now.", "I can't take it! Need a smoke NOW!")
+
+/datum/addiction/nicotine/withdrawal_enters_stage_1(mob/living/carbon/affected_carbon, delta_time)
+	. = ..()
+	affected_carbon.Jitter(5 * delta_time)
+
+/datum/addiction/nicotine/withdrawal_stage_2_process(mob/living/carbon/affected_carbon, delta_time)
+	. = ..()
+	affected_carbon.Jitter(10 * delta_time)
+	SEND_SIGNAL(affected_carbon, COMSIG_ADD_MOOD_EVENT, "depression_minimal", /datum/mood_event/depression_minimal)
```
Comment (Mothblocks) [inline code/modules/reagents/withdrawal/generic_addictions.dm:244]: This should be a custom moodlet to provide clearer feedback.

File: code/modules/reagents/withdrawal/generic_addictions.dm [hunk 3]
```diff
@@ -228,3 +228,26 @@
...
+	if(DT_PROB(10, delta_time))
+		affected_carbon.emote("cough")
+
+/datum/addiction/nicotine/withdrawal_stage_3_process(mob/living/carbon/affected_carbon, delta_time)
+	. = ..()
+	affected_carbon.Jitter(15 * delta_time)
+	SEND_SIGNAL(affected_carbon, COMSIG_ADD_MOOD_EVENT, "depression_moderate", /datum/mood_event/depression_moderate)
```
Comment (Mothblocks) [inline code/modules/reagents/withdrawal/generic_addictions.dm:251]: This too.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by maxymax13** [comment]
dreamseeker_G4vI6FYilD

**Pull request merged**